packages/serverless/scripts/build-awslambda-layer.js

```diff
@@ -0,0 +1,136 @@
+      // Internal deps aka local node_modules folder of each package is handled differently.
+      if (await pathExists(path.resolve(cwd, 'node_modules', dep))) {
+        return;
+      }
+
+      const searchPath = path.resolve(cwd, '..');
+      const depPath = await fs.realpath(
+        await findUp(path.join('node_modules', dep), { type: 'directory', cwd: searchPath }),
+      );
+      if (packages[dep]) {
+        if (packages[dep].cwd != depPath) {
+          throw new Error(`${packageJson.name}'s dependenciy ${dep} maps to both ${packages[dep].cwd} and ${depPath}`);
+        }
+        return;
+      }
+      await collectPackages(depPath, packages);
+    }),
+  );
+
+  return packages;
+}
+
+async function main() {
+  const workDir = path.resolve(__dirname, '..'); // packages/serverless directory
+  const packages = await collectPackages(workDir);
+
+  const dist = path.resolve(workDir, 'dist-awslambda-layer');
+  const destRootRelative = 'nodejs/node_modules/@sentry/serverless';
+  const destRoot = path.resolve(dist, destRootRelative);
+  const destModulesRoot = path.resolve(destRoot, 'node_modules');
+
+  await fs.remove(destRoot, { recursive: true });
+  await fs.mkdir(destRoot, { recursive: true });
+
+  await Promise.all(
+    Object.entries(packages).map(async ([name, pkg]) => {
+      const isRoot = name == '@sentry/serverless';
+      const destPath = isRoot ? destRoot : path.resolve(destModulesRoot, name);
+
+      // Scan over the distributable files of the module and symlink each of them.
+      const sourceFiles = await packList({ path: pkg.cwd });
+      await Promise.all(
+        sourceFiles.map(async filename => {
+          const sourceFilename = path.resolve(pkg.cwd, filename);
+          const destFilename = path.resolve(destPath, filename);
+
+          await fs.mkdir(path.dirname(destFilename), { recursive: true });
+          await fs.symlink(sourceFilename, destFilename);
+        }),
+      );
+
+      const sourceModulesRoot = path.resolve(pkg.cwd, 'node_modules');
+      if (!(await pathExists(sourceModulesRoot))) {
+        return;
+      }
+
+      // Scan over local node_modules folder of the package and symlink its non-dev dependencies.
+      const sourceModules = await fs.readdir(sourceModulesRoot);
+      await Promise.all(
+        sourceModules.map(async sourceModule => {
+          if (!pkg.packageJson.dependencies || !pkg.packageJson.dependencies[sourceModule]) {
+            return;
+          }
+
+          const sourceModulePath = path.resolve(sourceModulesRoot, sourceModule);
+          const destModulePath = path.resolve(destPath, 'node_modules', sourceModule);
+
+          await fs.mkdir(path.dirname(destModulePath), { recursive: true });
+          await fs.symlink(sourceModulePath, destModulePath);
+        }),
+      );
+    }),
+  );
+
+  const version = packages['@sentry/serverless'].packageJson.version;
```

Comment:
The `@sentry/serverless` name is used multiple times in the script. Maybe it could be worth it to extract it to the top, and use `require('../package.json').name` instead?